src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs

```diff
@@ -474,17 +482,21 @@ private void LoadLogs(ConsoleLogsSubscription newConsoleLogsSubscription)
                     foreach (var (lineNumber, content, isErrorOutput) in batch)
                     {
                         // Set the base line number using the reported line number of the first log line.
-                        if (_logEntries.EntriesCount == 0)
+                        _logEntries.BaseLineNumber ??= lineNumber;
+
+                        var logEntry = logParser.CreateLogEntry(content, isErrorOutput);
+                        if (logEntry.Timestamp is not null && timestampFilterDate is not null && !(logEntry.Timestamp > timestampFilterDate))
                         {
-                            _logEntries.BaseLineNumber = lineNumber;
+                            continue;
                         }
 
-                        var logEntry = logParser.CreateLogEntry(content, isErrorOutput);
-                        if (timestampFilterDate is null || logEntry.Timestamp is null || logEntry.Timestamp > timestampFilterDate)
+                        if (logEntry.Timestamp is not null && _logEntries.GetPauseEntries().Select(e => e.Pause).Cast<LogPauseViewModel>().FirstOrDefault(pause => pause.Contains(logEntry.Timestamp.Value)) is { } pause)
```

Comment:
If the user refreshes the page, may add several prior pauses to log entries. Each needs to be checked. I added a list of pause entries to LogEntries to speed up this call